src/controllers/EventsController.php

```diff
@@ -666,4 +684,33 @@ public function rejectAction($request, $db)
 
         return;
     }
+
+
+    public function setEventMapper(EventMapper $event_mapper)
+    {
+        $this->event_mapper = $event_mapper;
+    }
+
+    public function getEventMapper($db, $request)
+    {
+        if (! isset($this->event_mapper)) {
+            $this->event_mapper = new EventMapper($db, $request);
+        }
+
+        return $this->event_mapper;
+    }
+
+    public function setPendingTalkClaimMapper(PendingTalkClaimMapper $pending_talk_claim_mapper)
+    {
+        $this->pending_talk_claim_mapper = $pending_talk_claim_mapper;
+    }
+
+    public function getPendingTalkClaimMapper($db, $request)
+    {
+        if (! isset($this->pending_talk_claim_mapper)) {
+            $this->pending_talk_claim_mapper = new PendingTalkClaimMapper($db, $request);
+        }
+
+        return $this->pending_talk_claim_mapper;
+    }
```

Comment:
just an idea as I'm seeing a lot of setters and getters sprouting over the next commits: Can't we use Factory-Classes for getting the mappers? Either the default mapper is instantiated or you can call a static setter and then the set mapper will be returned. Basically moving the setMapper/getMapper into its own class.